### PR TITLE
font-new-computer-modern 6.0.0 (new cask)

### PR DESCRIPTION
**Pull request by peng1999**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/font/font-n/font-new-computer-modern.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  version "6.0.0"
+  sha256 "1013f36abab243e74ad892c443358896a6bdbc3e295286873e95c04cf8fe13c4"
+
+  url "https://download.gnu.org.ua/release/newcm/newcm-#{version}.txz"
+  name "New Computer Modern"
+  homepage "https://download.gnu.org.ua/release/newcm/"
+
+  livecheck do
+    url "https://download.gnu.org.ua/release/newcm/"
+    regex(/newcm-(\d+(?:\.\d+)+)\.txz/i)
```
Comment (khipp) [inline Casks/font/font-n/font-new-computer-modern.rb:11]:
```suggestion
    url "https://download.gnu.org.ua/release/newcm/"
    regex(/href=.*?newcm[._-]v?(\d+(?:\.\d+)+)\.t/i)
```

This updates the regex to align with our typical patterns (see [documentation](https://docs.brew.sh/Brew-Livecheck#file-names)).

File: Casks/font/font-n/font-new-computer-modern.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  font "newcm-6.0.0/otf/NewCMSans10-Book.otf"
+  font "newcm-6.0.0/otf/NewCMSans10-BookOblique.otf"
+  font "newcm-6.0.0/otf/NewCMSans10-Oblique.otf"
+  font "newcm-6.0.0/otf/NewCMSans10-Regular.otf"
+  font "newcm-6.0.0/otf/NewCMUncial08-Bold.otf"
+  font "newcm-6.0.0/otf/NewCMUncial08-Book.otf"
+  font "newcm-6.0.0/otf/NewCMUncial08-Regular.otf"
+  font "newcm-6.0.0/otf/NewCMUncial10-Bold.otf"
+  font "newcm-6.0.0/otf/NewCMUncial10-Book.otf"
+  font "newcm-6.0.0/otf/NewCMUncial10-Regular.otf"
```
Comment (khipp) [inline Casks/font/font-n/font-new-computer-modern.rb:53]:
```suggestion
  font "newcm-#{version}/otf/NewCM08-Book.otf"
  font "newcm-#{version}/otf/NewCM08-BookItalic.otf"
  font "newcm-#{version}/otf/NewCM08-Italic.otf"
  font "newcm-#{version}/otf/NewCM08-Regular.otf"
  font "newcm-#{version}/otf/NewCM08Devanagari-Book.otf"
  font "newcm-#{version}/otf/NewCM08Devanagari-Regular.otf"
  font "newcm-#{version}/otf/NewCM10-Bold.otf"
  font "newcm-#{version}/otf/NewCM10-BoldItalic.otf"
  font "newcm-#{version}/otf/NewCM10-Book.otf"
  font "newcm-#{version}/otf/NewCM10-BookItalic.otf"
  font "newcm-#{version}/otf/NewCM10-Italic.otf"
  font "newcm-#{version}/otf/NewCM10-Regular.otf"
  font "newcm-#{version}/otf/NewCM10Devanagari-Bold.otf"
  font "newcm-#{version}/otf/NewCM10Devanagari-Book.otf"
  font "newcm-#{version}/otf/NewCM10Devanagari-Regular.otf"
  font "newcm-#{version}/otf/NewCMMath-Bold.otf"
  font "newcm-#{version}/otf/NewCMMath-Book.otf"
  font "newcm-#{version}/otf/NewCMMath-Regular.otf"
  font "newcm-#{version}/otf/NewCMMono10-Bold.otf"
  font "newcm-#{version}/otf/NewCMMono10-BoldOblique.otf"
  font "newcm-#{version}/otf/NewCMMono10-Book.otf"
  font "newcm-#{version}/otf/NewCMMono10-BookItalic.otf"
  font "newcm-#{version}/otf/NewCMMono10-Italic.otf"
  font "newcm-#{version}/otf/NewCMMono10-Regular.otf"
  font "newcm-#{version}/otf/NewCMSans08-Book.otf"
  font "newcm-#{version}/otf/NewCMSans08-BookOblique.otf"
  font "newcm-#{version}/otf/NewCMSans08-Oblique.otf"
  font "newcm-#{version}/otf/NewCMSans08-Regular.otf"
  font "newcm-#{version}/otf/NewCMSans10-Bold.otf"
  font "newcm-#{version}/otf/NewCMSans10-BoldOblique.otf"
  font "newcm-#{version}/otf/NewCMSans10-Book.otf"
  font "newcm-#{version}/otf/NewCMSans10-BookOblique.otf"
  font "newcm-#{version}/otf/NewCMSans10-Oblique.otf"
  font "newcm-#{version}/otf/NewCMSans10-Regular.otf"
  font "newcm-#{version}/otf/NewCMUncial08-Bold.otf"
  font "newcm-#{version}/otf/NewCMUncial08-Book.otf"
  font "newcm-#{version}/otf/NewCMUncial08-Regular.otf"
  font "newcm-#{version}/otf/NewCMUncial10-Bold.otf"
  font "newcm-#{version}/otf/NewCMUncial10-Book.otf"
  font "newcm-#{version}/otf/NewCMUncial10-Regular.otf"
```

We should interpolate the `version` to require fewer changes when updating the cask.

File: Casks/font/font-n/font-new-computer-modern.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+cask "font-new-computer-modern" do
+  version "6.0.0"
+  sha256 "1013f36abab243e74ad892c443358896a6bdbc3e295286873e95c04cf8fe13c4"
+
+  url "https://download.gnu.org.ua/release/newcm/newcm-#{version}.txz"
+  name "New Computer Modern"
+  homepage "https://download.gnu.org.ua/release/newcm/"
```
Comment (khipp) [inline Casks/font/font-n/font-new-computer-modern.rb:7]:
```suggestion
  homepage "https://git.gnu.org.ua/newcm.git/about/"
```

It seems like the font has no dedicated website, but the repository might still provide more information.

**Review by khipp — COMMENTED** [review]
Thank you, @peng1999! I left a few minor suggestions below.

**Review by bevanjkay — APPROVED** [review]

**Review by khipp — APPROVED** [review]

**Pull request merged**